pkg/spec/spec.go

```diff
@@ -118,10 +119,20 @@ func GroupFromBytes(b []byte) (*Group, error) {
 			// unmarshal to a raw map as we don't want to strip away unknown/invalid
 			// fields at this point - that validation is done in the apply spec/teams
 			// endpoint so that it is enforced for both the API and the CLI.
-			rawTeam := make(map[string]json.RawMessage)
+			rawTeam := make(map[string]map[string]json.RawMessage)
 			if err := yaml.Unmarshal(s.Spec, &rawTeam); err != nil {
 				return nil, fmt.Errorf("unmarshaling %s spec: %w", kind, err)
 			}
+			// special-case the agent_options key: we want to leave the team's agent
```

Comment:
This introduces some interesting differences between using fleetctl to apply a yaml spec, and calling the json api endpoint `POST /fleet/spec/teams`. When using yaml, `null` turns into `{}`. When calling the json api endpoint directly, `null` stays as `null` and the agent options are not updated.

Reply:
We can differentiate between different cases for `agent_options` in both yaml and json by changing the type of `AgentOptions` from `*json.RawMessage` to `json.RawMessage` ie don't use a pointer. In the following

```
type TeamSpec struct {
	Name         string           `json:"name"`
	AgentOptions json.RawMessage  `json:"agent_options"`
```

Here are the following scenarios when decoding

1. `agent_options` is missing -> `AgentOptions` is nil
2. `agent_options` is `null` -> `AgentOptions` is `json.RawMessage("null")`
3. `agent_otpions` is `{...}` -> `AgentOtpions` is `json.RawMessage("{...}")`

Reply:
Oh that's true, I like that idea! I'll give it a shot, thanks!